gramps/plugins/textreport/detancestralreport.py

```diff
@@ -405,7 +406,7 @@ def write_more_header(first, name):
                 self.doc.start_paragraph('DAR-MoreDetails')
 
                 text = utils.get_address_str(addr)
-                self.doc.write_text(self._('Address: '))
+                self.doc.write_text(addr.get_schema()["title"] + ': ')
```

Comment:
Good idea, but I don't think that it will work with deferred translations in reports.  Also, you can't simply concatenate ': ' because the French will want a space before the colon.

Reply:
You say about this:
                self.doc.write_text(addr.get_schema()["title"] + _(': '))

Reply:
No, this is incorrect. This is called "a string puzzle". It doesn't work in RTL languages for  example.

Please check what is written here: it is from the KDE community, but the basic principles of the "Writing Good Texts" section applies to all translations, regardless of the syntax:
https://api.kde.org/frameworks/ki18n/html/prg_guide.html#good_text

Reply:
You're right. I forgot about RTL.
It's must be:
`                self.doc.write_text(_('{address}: ').format(address=addr.get_schema()["title"]))
`

Reply:
This still doesn't work with deferred translations.  Test by selecting a different "Translation" in "Report Options (2)".

Reply:
For work in other locale "Report Options (2)" must send additional locale to 'get_schema()':
```
                self.doc.write_text(self._('{address}: ').format(address=addr.get_schema(self._locale)["title"]))
```

And /address.py.get_schema() must have additional parameter and code:
```
    def get_schema(cls, s_locale):
...
        from .date import Date

        new_lang = s_locale.translation
        new_lang.install()
        _=new_lang.gettext

        return {
            "type": "object",
            "title": _("Address"),
            "properties": {
```
Many .py files in '/gramps/gen/lib/' have schemas. All this files can be used with additional locale for creating reports in any supported language. Now this schema can be used in default language only.

But it's must be one big PR with many changed files...
It's need?